axelrod/tournament.py

```diff
@@ -307,20 +313,22 @@ def _process_done_queue(self, workers: int, done_queue: Queue,
             The number of sub-processes in existence
         done_queue : multiprocessing.Queue
             A queue containing the output dictionaries from each round robin
-        progress_bar : bool
-            Whether or not to update the tournament progress bar
         """
+        file, writer = self._get_file_objects()
```

Comment:
it could be done at the start and was originally done that way.  I doubt that performance would be affected more than a few nanoseconds either way (extra overhead for separate method call vs. return obj and passing to various methods.).  the same check is still being performed exactly once, just in different places.  I prefer the current way, because the original involved passing a `writer` variable through all the methods which had the variable for the sole purpose of passing it to the next method.  It seemed cleaner to open and close the file closest to where it was needed.  

original:
- `play`
    - call `setup_output` and (possibly) create temp file
       - still needs to check `filename` to see if file is created
       - create writer and file obj
       - return writer and file obj
    - call `_run_paralell(writer=writer)`
        - call `_start_workers`
        - call `_process_done_queue(writer=writer)`
          - call `_write_interactions(writer=writer)`
    - close file object
    - delete temp file

newer:
- `play`
    - call `setup_output` and (possibly) create temp file
    - call `_run_paralell`
        - call `_start_workers`
        - call `_process_done_queue`
          - check for file creation
          - create file obj and csv writer
          - call `_write_interactions(writer=writer)`
          - close file object
    - delete temp file